### PR TITLE
misc changes

### PR DESCRIPTION
**Pull request by gibix**
- Updated copyright
- Updated cargo to 0.27 with apis (dropped call to `call_main_without_stdin`) and new cargo's apis
- Add quicli to manage cli interafce

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,6 @@ rust:
 - nightly
 - beta
 - stable
-- 1.17.0 # minimum supported version (due to unicode-bidi breakage)
```
Comment (cardoe) [inline .travis.yml]: Why?

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "cargo-ebuild"
-version = "0.1.6-pre"
+version = "0.1.7"
```
Comment (cardoe) [inline Cargo.toml]: I use `cargo-release` so just leave this be.

File: Cargo.toml [hunk 3]
```diff
@@ -13,11 +13,15 @@ description = """
 Generates an ebuild for a package using the in-tree eclasses.
 """
 
-[badges]
-travis-ci = { repository = "cardoe/cargo-ebuild" }
+[badges.travis-ci]
+repository = "cardoe/cargo-bitbake"
```
Comment (cardoe) [inline Cargo.toml]: This can't be right. I think I have an idea where you copied this from... ;)

File: src/main.rs [hunk 4]
```diff
@@ -120,19 +86,18 @@ fn real_main(options: Options, config: &Config) -> CliResult {
...
 
     // package description
-    let desc = metadata
-        .description
-        .as_ref()
-        .cloned()
-        .unwrap_or_else(|| String::from(package.name()));
+    let desc = metadata.description.as_ref().cloned().unwrap();
+    // TODO: now package.name is InternedString
+    // .unwrap_or_else(|| String::from(package.name()));
```
Comment (cardoe) [inline src/main.rs]: Yeah we'll need to resolve this prior to merge.

File: src/main.rs [hunk 5]
```diff
@@ -144,26 +109,26 @@ fn real_main(options: Options, config: &Config) -> CliResult {
...
-                            .create(true)
-                            .truncate(true)
-                            .open(&ebuild_path)
-                            .chain_err(|| "failed to create ebuild"));
+    let mut file = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .truncate(true)
+        .open(&ebuild_path)
+        .unwrap();
```
Comment (cardoe) [inline src/main.rs]: I wouldn't want this to panic when you failed to open the file. Returning an error saying it couldn't create the file would be nicer.

File: src/main.rs [hunk 6]
```diff
@@ -144,26 +109,26 @@ fn real_main(options: Options, config: &Config) -> CliResult {
...
+    write!(
+        file,
+        include_str!("ebuild.template"),
+        description = desc.trim(),
+        homepage = homepage.trim(),
+        license = license.trim(),
+        crates = crates.join(""),
+        cargo_ebuild_ver = env!("CARGO_PKG_VERSION"),
+        this_year = 1900 + time::now().tm_year,
+    ).expect("Error during ebuild file writing");
```
Comment (cardoe) [inline src/main.rs]: Again this is going to panic instead of returning a nice error to the user.

File: .travis.yml (hunk 1 above)
Comment (gibix) [reply .travis.yml]: now 1.17 is nightly

File: Cargo.toml (hunk 3 above)
Comment (gibix) [reply Cargo.toml]: yep

**Comment by gibix** [comment]
with the last two commit there is a bit of refactoring and better error handling

**Comment by gibix** [comment]
add basic git crates support

better errors with failure
better cli
support for env debut
small fix

File: .travis.yml (hunk 1 above)
Comment (cardoe) [reply .travis.yml]: No. That's a stable release.

**Comment by gibix** [comment]
@cardoe nothing will never compile for 1.17. I understand the broken unicode-bidi release but why we need an expicit test on 1.17?

**Comment by gibix** [comment]
This is a single, quite big, refactor change. Me can squash it with a better message and changelog. Something has been rewritten/reverted.

**Pull request closed without merging**